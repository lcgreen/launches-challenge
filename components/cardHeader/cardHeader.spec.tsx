import React from 'react';
import { render, screen } from '@testing-library/react';
import CardHeader, { CardHeaderProps } from '.';

const defaultProps: CardHeaderProps = {
  title: 'Title',
  image: 'https://via.placeholder.com/150',
  imageAlt: 'Image Alt',
  imageWidth: 150,
  imageHeight: 150,
  utcDate: '2021-08-01T00:00:00.000Z',
  status: 'success',
};

describe('CardHeader', () => {
  it('renders the component', async () => {
    const { container } = render(<CardHeader {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the title', async () => {
    render(<CardHeader {...defaultProps} />);
    expect(screen.getByText('Title')).not.toBeNull();
  });

  it('renders the image', async () => {
    render(<CardHeader {...defaultProps} />);
    expect(screen.getByAltText('Image Alt')).not.toBeNull();
  });

  it('renders the date', async () => {
    render(<CardHeader {...defaultProps} />);
    expect(screen.getByText('Sun Aug 01 2021')).not.toBeNull();
  });
});
