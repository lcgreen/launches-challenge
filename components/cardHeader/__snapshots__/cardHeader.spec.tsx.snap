// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader renders the component 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="Image Alt"
      class="image"
      data-nimg="1"
      decoding="async"
      height="150"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F150&w=384&q=75 2x"
      style="color: transparent;"
      width="150"
    />
    <div
      class="body"
    >
      <div
        class="title"
      >
        Title
      </div>
      <span
        class="date"
      >
        Sun Aug 01 2021
      </span>
    </div>
  </div>
</div>
`;
